subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/ChangesDuringTheBuildFileSystemWatchingIntegrationTest.groovy

```diff
@@ -47,32 +39,9 @@ class ChangesDuringTheBuildFileSystemWatchingIntegrationTest extends AbstractFil
                     ${server.callFromBuild("userInput")}
                 }
             }
-
-            gradle.buildFinished {
-                def projectRoot = project.projectDir.absolutePath
-                def vfs = gradle.services.get(VirtualFileSystem)
-                int filesInVfs = 0
-                vfs.getRoot().visitSnapshotRoots { snapshot ->
-                    snapshot.accept(new FileSystemSnapshotVisitor() {
-                        @Override
-                        void visitFile(CompleteFileSystemLocationSnapshot fileSnapshot) {
-                            if (fileSnapshot.type == FileType.RegularFile && fileSnapshot.absolutePath.startsWith(projectRoot)) {
```

Comment:
The point of this `buildFinished` hook is to capture the files in the VFS from the project since we know which ones those are. We also keep files from the global caches in the VFS after the build, and that made it harder to do the assertions, especially since the number seems to change based on some conditions. So I really would prefer to keep this hook.

Reply:
Let's discuss this separately.